dns-controller/pkg/dns/dnscontroller.go

```diff
@@ -480,19 +480,42 @@ func (s *DNSControllerScope) MarkReady() {
 }
 
 func (s *DNSControllerScope) Replace(recordName string, records []Record) {
-	glog.V(2).Infof("Update %s/%s: %v", s.ScopeName, recordName, records)
-
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
+	existing, exists := s.Records[recordName]
+
 	if len(records) == 0 {
+		if !exists {
+			return
+		}
+
 		delete(s.Records, recordName)
 	} else {
+		if recordsSliceEquals(existing, records) {
+			return
```

Comment:
We are returning a couple of times and not logging.  Would some debug logging help?